src/server/JasmineGraphServer.cpp

```diff
@@ -2767,77 +1856,79 @@ void JasmineGraphServer::outDegreeDistribution(std::string graphID) {
-                    server_logger.log("Error writing to socket", "error");
-                }
-                server_logger.log("Sent : Partition ID " + std::to_string(partitionID), "info");
+        response = Utils::read_str_trim_wrapper(sockfd, data, FED_DATA_LENGTH);
+        if (response.compare(JasmineGraphInstanceProtocol::OK) != 0) {
+            server_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::OK +
+                                " ; Received: " + response);
+            close(sockfd);
+            continue;
+        }
+        server_logger.info("Received: " + response);
 
-                bzero(data, FED_DATA_LENGTH + 1);
-                read(sockfd, data, FED_DATA_LENGTH);
-                string response = (data);
-                response = Utils::trim_copy(response, " \f\n\r\t\v");
+        if (!Utils::send_str_wrapper(sockfd, partition)) {
+            close(sockfd);
+            continue;
+        }
+        server_logger.info("Sent: " + partition);
 
-                if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
-                    server_logger.log("Received : " + JasmineGraphInstanceProtocol::OK, "info");
-                    result_wr = write(sockfd, workerList.c_str(), workerList.size());
+        response = Utils::read_str_trim_wrapper(sockfd, data, FED_DATA_LENGTH);
+        if (response.compare(JasmineGraphInstanceProtocol::OK) != 0) {
+            server_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::OK +
+                                " ; Received: " + response);
+            close(sockfd);
+            continue;
+        }
+        server_logger.info("Received: " + response);
 
-                    if (result_wr < 0) {
-                        server_logger.log("Error writing to socket", "error");
-                    }
-                    server_logger.log("Sent : Host List ", "info");
-                }
-            }
+        if (!Utils::send_str_wrapper(sockfd, workerList)) {
+            close(sockfd);
+            continue;
         }
+        server_logger.info("Sent: " + workerList);
     }
 }
 
+void JasmineGraphServer::inDegreeDistribution(std::string graphID) {
+    degreeDistributionCommon(graphID, JasmineGraphInstanceProtocol::IN_DEGREE_DISTRIBUTION);
+}
+
+void JasmineGraphServer::outDegreeDistribution(std::string graphID) {
+    degreeDistributionCommon(graphID, JasmineGraphInstanceProtocol::OUT_DEGREE_DISTRIBUTION);
+}
+
 long JasmineGraphServer::getGraphVertexCount(std::string graphID) {
     auto *refToSqlite = new SQLiteDBInterface();
     refToSqlite->init();
     vector<vector<pair<string, string>>> output =
         refToSqlite->runSelect("SELECT vertexcount FROM graph WHERE idgraph = '" + graphID + "'");
-
-    long vertexCount = std::stoi(output[0][0].second);
     refToSqlite->finalize();
     delete refToSqlite;
+
+    long vertexCount = std::stoi(output[0][0].second);
```

Comment:
Can there be a chance for output[0][0] to be null?

Reply:
Yes, if the graph ID is invalid. But there are many such missing validations which needs to be fixed separately.

Reply:
Created Issue https://github.com/miyurud/jasminegraph/issues/233 to handle this